vizro-core/tests/unit/vizro/tables/test_dash_ag_grid.py

```diff
@@ -30,3 +32,21 @@ def test_dash_ag_grid(self):
         )
         # we could test other properties such as defaultColDef,
         # but this would just test our chosen defaults, and no functionality really
+
+
+class TestCustomDashAgGrid:
+    def test_custom_dash_ag_grid(self):
+        @capture("ag_grid")
+        def custom_ag_grid(data_frame):
+            return dag.AgGrid(
+                columnDefs=[{"field": col} for col in data_frame.columns],
+                rowData=data_frame.to_dict("records"),
+            )
+
+        grid_model = vm.AgGrid(
+            id="custom_ag_grid",
+            title="Custom Dash AgGrid",
+            figure=custom_ag_grid(data_frame=data),
+        )
+        grid_model.pre_build()
+        grid_model.build()
```

Comment:
This tests reads pretty weirdly because there's no `assert` (same goes for the Dash data table one). I think we should one or both of these:
1. add an `assert` e.g. `assert_component_equal` on the results of the `build` or just check the output of that has the right `id`
2. put in some comment explaining what this is actually testing

Reply:
Did both :)